sdk/personalizer/Azure.AI.Personalizer/src/Models/PersonalizerClient.cs

```diff
@@ -548,34 +549,62 @@ public virtual Response ActivateMultiSlot(string eventId, CancellationToken canc
             }
+            personalizerServiceProperties = ServiceConfigurationRestClient.Get(cancellationToken);
+            personalizerPolicy = PolicyRestClient.Get(cancellationToken);
+            //interactions & observations
             config["interaction.http.api.host"] = stringEndpoint + "personalizer/v1.1-preview.2/logs/interactions";
             config["observation.http.api.host"] = stringEndpoint + "personalizer/v1.1-preview.2/logs/observations";
-            config["interaction.subsample.rate"] = Convert.ToString(subsampleRate, CultureInfo.InvariantCulture);
-            config["observation.subsample.rate"] = Convert.ToString(subsampleRate, CultureInfo.InvariantCulture);
-            //ToDo: TASK 13057958 Working on changes to support model api in RL.Net
+            config["interaction.sender.implementation"] = "INTERACTION_HTTP_API_SENDER";
+            config["observation.sender.implementation"] = "OBSERVATION_HTTP_API_SENDER";
+            config["interaction.subsample.rate"] = Convert.ToString(this.subsampleRate, CultureInfo.InvariantCulture);
+            config["observation.subsample.rate"] = Convert.ToString(this.subsampleRate, CultureInfo.InvariantCulture);
+            //model
             config["model.blob.uri"] = stringEndpoint + "personalizer/v1.1-preview.1/model";
-            config["vw.commandline"] = _personalizerPolicy.Arguments;
+            config["model.source"] = "HTTP_MODEL_DATA";
+
+            config["model.vw.initial_command_line"] = personalizerPolicy.Arguments;
             config["protocol.version"] = "2";
-            config["initial_exploration.epsilon"] = Convert.ToString(_personalizerServiceProperties.ExplorationPercentage, CultureInfo.InvariantCulture);
-            config["rank.learning.mode"] = Convert.ToString(_personalizerServiceProperties.LearningMode, CultureInfo.InvariantCulture);
-            //return the config model
-            return config;
+            config["initial_exploration.epsilon"] = Convert.ToString(personalizerServiceProperties.ExplorationPercentage, CultureInfo.InvariantCulture);
+            config["rank.learning.mode"] = Convert.ToString(personalizerServiceProperties.LearningMode, CultureInfo.InvariantCulture);
+
+            LiveModel liveModel = new LiveModel(config);
+            liveModel.Init();
+            liveModelLastRefresh = DateTimeOffset.UtcNow;
+            return new RankProcessor(liveModel);
+        }
+
+        /// <summary> Update the config details periodically based on liveModelRefreshTimeInMinutes or when bearer token is expired </summary>
+        internal void validateAndUpdateLiveModelConfig()
+        {
+            if ((tokenCredential != null &&
+                DateTimeOffset.Compare(tokenExpiry, DateTimeOffset.MinValue) != 0 &&
+                DateTimeOffset.Compare(tokenExpiry, DateTimeOffset.UtcNow) <= 0) ||
```

Comment:
Token expiry need to be a couple minutes away from now.

Reply:
until the token expiry the same token is generated. So even with the buffer of 1 min unless the token is expired new token won't be generated with the new expiry. That's what I understood